toolkit/scripts/imggen.mk

```diff
@@ -5,7 +5,8 @@ $(call create_folder,$(IMAGEGEN_DIR))
 
 # Resources
 config_name              = $(notdir $(CONFIG_FILE:%.json=%))
-config_other_files       = $(if $(CONFIG_FILE),$(call shell_real_build_only, find $(CONFIG_BASE_DIR)))
+config_dir_name          = $(dir $(CONFIG_FILE:%.json=%))
+config_other_files       = $(if $(CONFIG_FILE),$(call shell_real_build_only, find $(config_dir_name) -type f -name $(config_name)))
```

Comment:
This will not capture things like the `my_postinstall.sh` which might be next to the config.json. I think the more robust solution here is to use `jq` or a similar tool to parse the config, and gather a list of files to monitor. (install scripts, additional files, package lists, etc.).